### DISCUSSION TITLE
Quick update on the current status of Actual

### DISCUSSION THREAD

**Discussion by jlongster** [actualbudget/releases#19] [Announcements]
Hello! I'm experimenting with using github discussions as place to communicate more permanent information to the community. Typically I used the [blog](https://actualbudget.com/blog/) for that, but something more forum-like is more conducive to, well, discussions. Blog posts are too formal and take too long to write. I like the informality here and allowing anybody to comment or even create their own discussion.

YNAB made some... interesting changes recently, and as a result a lot of people are looking at Actual. I'm excited about that! I wanted to give a super quick run down on some current bugs that you might be hitting.

I will make another post soon about the longer-term vision of Actual, and give historical context for it and tell more about about myself (hi I'm James). The short of it is:

* Actual has been around for many years, is going to stick around for many more, and is not just "another YNAB4" but in fact has much larger plans like the ability to write custom reports with our own query language [AQL](https://actualbudget.com/docs/developers/ActualQL/).

* It already supports a large amount of features such as zero-based budgeting, payees, CSV/OFX/QSF importing, a web/desktop/mobile app, transfers, undo/redo, end-to-end encryption, and more.

* Two unique features are rules and schedules. It doesn't just schedule transactions, but it tracks the status of all your recurring transactions and gives a nice overview like this:

<img width="1132" alt="Screen Shot 2021-11-02 at 10 34 22 PM" src="https://user-images.githubusercontent.com/17031/140002603-1cc6bb96-4575-4d77-9528-c2caecba901f.png">

You can watch a video about scheduled transactions [here](https://actualbudget.com/release-notes/#0.0.133).

* Custom rules allow you create any kind of workflow you like. You can match transactions based on a set of rules, and apply any actions to them.

# Current bugs in 0.0.141

The current version is 0.0.141. Unfortunately, this release made a _huge_ change internally which fixes a few critical issues (short version: how caching works was completely revamped making it more stable). While the release was successful, there are a few critical bugs that weren't being reported, so I was unaware of them until a bunch of new users tried the app.

## Not a cloud file / could not upload file

New users may see this when creating a new file. In Actual, all data is local, and what happens here is the local file is created but there's an error trying to upload it and "register" it with the server.

This happened on the web and mobile version in 0.0.141, but not on the desktop app. I have already pushed out 0.0.142 to fix this on the web, so **currently both the web and desktop app should work**, but the mobile app still has this problem. I'm working on officially releasing 0.0.142, but for the mobile apps I have to wait for them to be reviewed.

The fix is to log into the [web app](http://app.actualbudget.com) and start a new file there. Then load that file on the mobile app. If you've already created a local file on mobile, go to settings and press "close file" to close and you should see the file you created from the web app available. It should be able to sync.

## YNAB4 import broken

Sadly, a code path was missed in the YNAB4 importer that was using something in the database which was removed in the latest version. That means YNAB4 importing right now doesn't work.

I know I know, bad timing right?

I will get this fixed asap in the official 0.0.142 release.

**Comment by jlongster** [comment]
Oh also:

# Importing from nYNAB

We wanted the community to help build the importer and made a starting point [here](https://github.com/actualbudget/import-ynab5). This does not fully work yet, however, but somebody else has flesh it out into a full importer. You can download it here:

[import-ynab5-built-on-aon.zip](https://github.com/actualbudget/releases/files/7465014/import-ynab5-built-on-aon.zip)

This is **not official** and you should **run this carefully**. Look at the source code first. It has worked for other users, however, and we plan on merging in these changes to the official importer.

From someone on [slack](https://slack.actualbudget.com):

```
Get an authorization token/API key/whatever it was called.
https://app.youneedabudget.com/settings/developer

Get the budgets list:
curl -H "Authorization: Bearer <yourkeyhere>" https://api.youneedabudget.com/v1/budgets

Get the account transactions:
curl -H "Authorization: Bearer <yourkeyhere>" https://api.youneedabudget.com/v1/budgets/YOURACCOUNTID > YourBudget.json

Import to Actual:
node index.js YourBudget.json
```

**Reply by MikeBishop** [reply]
But this also will require the 142 release, correct?

**Reply by jlongster** [reply]
Good point, it might, not sure if it uses one of the API methods that hits the bug. I'm about to release it today though regardless.

**Reply by jlongster** [reply]
142 is the latest release now, everyone should be good!

**Reply by stefer09** [reply]
I'm trying to jump ship from nYNAB to Actual, ran the import node script, everything seems to import fine, except... a LOT of categories.  A lot of that list should not have a category anyway (at least that's how it worked in YNAB, for credit-card payments, or transfer between accounts), but quite a few are wrong, like any income are not categorized.

**Reply by Silvenga** [reply]
During my migration, I saw:
- Income wasn't categorized - that one is rather easy to fix.
- Any split transactions were added in as a single transaction without a category. I only had a handful so it wasn't bad.
- All transfers weren't mapped up correct - lot's of manual work getting that in - but Actual is so much faster than YNAB for bulk operations.
- Scheduled transactions were added as transactions based on the create time (IIRC) - that one was really hard to track down.
- Scheduled transactions had to be manually added again.

Ultimately it wasn't bad - about 6 years of transactions (15MB of json) - fixed after a couple of hours of work.

**Reply by stefer09** [reply]
For splits, I have a LOT, so I might be better off starting a fresh, new budget...

**Reply by darek** [reply]
Got problem because non of my "budgeted" from "moths" was imported

**Reply by Silvenga** [reply]
Another odd problem (caching?), after import my budgets had no "budgeted" - so I manually fixed the last month. After syncing to another machine, all the budgeted amounts showed up from the import (meaning my last month was now completely wrong).

**Reply by Mreco99** [reply]
I jumped ship from YNAB for two reasons
1. YNAB is too expensive, please dont go down that road. 
2. Your software fits my usage brilliantly, by all means add features but not at the cost of price.

**Comment by chaorace** [comment]
Apologies if this has already been recently addressed: what's the current state of development regarding bank imports?

**Reply by chaorace** [reply]
To clarify: I mean in the context of ongoing transaction synchronization

**Reply by Silvenga** [reply]
Yes! I want to know if it's worth rolling my own for now, or if I should just wait. 

(I hope the mass YNAB migration isn't adding stress)

**Comment by stephmeyer** [comment]
Just watched your Twitch video.  Thanks for being so open and forthright!

Ah, I was excited about the YNAB4 importing and imported my file (I just started using YNAB4 and set up my budget and was going to try Actual as a comparison, side by side, for awhile), and yeah, it totally didn't work.  It imported my accounts but none of the money is in available to budget and the budgeted amounts also aren't there.   I might just use YNAB4 for awhile and wait till that is fixed, so please keep updates coming.  I don't want to re-do everything I just set up in YNAB4.

Did you figure out how to get YNAB4 working on your computer?  I got it working on my Mac with an M1 chip.  I can email you the details if you haven't gotten it working yet.  

I am so excited to see a real person instead of a soul-less company working on a budgeting app that can hopefully soon compete with a certain company that we used to trust and now don't want to give a dime to.

 I do think the importing of old debt as it's own category (actually a full category group with a category for each card) in YNAB4 was nice and you might want to consider something like that happening automatically as a priority in Actual development, especially now that you're hopefully soon going to be getting an influx of the people who can't afford nYNAB's price and may need to slowly pay off that old debt, and that's why they're actually budgeting.  Right now I think you may be getting the old timer legacy YNAB users coming over, who probably aren't carrying credit card balances, but if you want to keep pushing the app forward, you'll need a way to help people get out of cc debt.  I'd guess a large proportion of the people who start budgeting are doing so to help them get out of cc debt or other kinds of debt, so if you want to attract newbie budgeters, dealing with old cc debt in a way that is intuitive is key.

I hope you can quit your other job and focus on Actual full time soon, with all these YNAB converts coming over!  Your app is a sorely needed competition to YNAB and I really appreciate the work you've put into it so far.  Please don't lose your soul and start to treat your early supporters like garbage in 10 years!!!!   I'm promoting it on all the forums where legacy YNABers are upset about how they have been treated by YNAB, since IMO it looks like the most viable competition to implement the 4 rules without the YNAB software.  I appreciate how YNAB helped me get where I am today, but I'd rather support your project than YNAB, if only because YNAB being an effective monopoly is bad for everyone.  Just get the YNAB4 import working please :)  That will help you attract some of the folks who have gone back to YNAB4; I think some of them will eventually miss having more features on the app and eventually it will break anyways.  Plus it's already got some weirdness for me.  For example, if I open YNAB4 with my external monitor plugged in it renders all funky.  I have to unplug my monitor for it to look right and I can only use it on my actual laptop screen.

Sorry so long winded.  Thanks for all your efforts and I wish you much success!!!!!

**Comment by alexmidd** [comment]
Hi and thanks for the update.

I have been trying Actual for a few days and am still getting used to it. One of the things that is a little frustrating is in relation to scheduled transactions - there seems to be no way to assign them to a budget category until after it has gone into the account register, and even then I couldn’t get it to do it. I clicked on “Due” but nothing would happen. 

For me, it should be assignable to a category from the outset as for one thing it saves time reassigning each transaction after it gets registered, but also it shows what the upcoming liabilities are so that you can budget for them correctly. Is this the intended behaviour, or have I just not been using it properly?

Alex

**Reply by bdoherty** [reply]
@alexmidd Have you tried clicking on the 'Edit as rule' button, and choosing "set category " as an action?

![image](https://user-images.githubusercontent.com/335468/142040466-8c5d3047-b38e-4d09-bb15-befb9662672a.png)

![image](https://user-images.githubusercontent.com/335468/142040599-4386a2ee-6503-47b0-82c6-f33ed3d356cc.png)

**Reply by alexmidd** [reply]
Hi bdoherty, thanks for the reply.

Yes, I've now discovered that option and tested it out and it seems to work fine. It's slightly different from other, similar applications but it does the same thing, essentially, which is fine.

Thanks

Alex

**Reply by Deguello** [reply]
This works.

However, it's extra steps and it obfuscates which Schedules really do have custom conditions and actions when the only "custom" thing is adding a category.

Maybe there's a design or functionality reason I'm just missing, but Category is, it seems to me, a fundamental property of every transaction and it should be settable as a default property.

**Reply by bpaulien** [reply]
I agree.  I have gone through and set up a bunch of my scheduled transactions that I had in YNAB, and it was a pain to create the schedule, save it, then go back into the schedule, "Edit as rule" and then add the category (and note), then you have 2 "save" buttons to hit before you're done. 

Much simpler to just have category (and possibly note?) in the initial screen for creating scheduled transactions.

**Reply by MikeBishop** [reply]
On the flip side, Actual does auto-generate rules that assign categories based on Payee, so after setting the Category a time or two, the rule should automatically apply it when the transaction posts.

**Reply by Deguello** [reply]
It's nice that it learns to associate a category with a payee, but I'm not sure how that's easier or better than simply assigning the category on the Schedule, especially when it's not crazy that different payments to the same payee can have different categories.

Edit: And a Schedule could be a split, which would be much easier to handle at creation, imo.

**Reply by bdoherty** [reply]
Raised Issue https://github.com/actualbudget/releases/issues/74

**Reply by alexmidd** [reply]
Another thing I have noticed is that even though you can set a rule to set the category, you can't set a split category. Or you can but there's no way that I can see that the amounts of the split can be set. To me these are basic things that would need to be set in any budgeting app.
